### PR TITLE
Request accept buttons into dropdown

### PR DESCRIPTION
**Pull request by ncounter**
The proposal is about keeping separated all the different decision actions, but grouping the `accept request` ones. Accepting request can be made like:

1. Accept
2. Accept & make the submitter a maintainer of the target project/package

Due to the fact that the forward can be submitted to multiple choices, they requires checkboxes to be visible and selectable. The forwarding destinations are shown as checkboxes above the buttons

image

### CODE REVIEW

File: src/api/app/components/request_decision_component.html.haml [hunk 1]
```diff
@@ -1,33 +1,97 @@
...
+                = submit_tag('Accept and forward request to the selected projects ', name: 'accepted', class: 'btn-link dropdown-item',
+                              data: confirmation, title: 'Accept the request. this will apply the changes on the target.',
+                              id: 'accept-and-forward')
+                -# Accept and make maintainer and forward submit
+                - if show_add_submitter_as_maintainer_option?
+                  = submit_tag("Accept and make #{@bs_request.creator} maintainer of #{make_maintainer_of} and forward request to the selected projects",
+                                name: 'accepted', class: 'btn-link dropdown-item', data: confirmation,
+                                title: 'Accept the request. this will apply the changes on the target.',
+                                id: 'accept-and-make-maintainer-and-forward')
+
```
Comment (hellcp-work) [inline src/api/app/components/request_decision_component.html.haml:84]: Do you need a second dropdown? The only visible difference between the two dropdowns are the checkboxes, which could just be integrated into the first dropdown instead

**Review by rubhanazeem — APPROVED** [review]

**Comment by danidoni** [comment]
> Adding a quick note - we need a followup on this to add confirmation dialog on accepting a request

This will be fixed here https://trello.com/c/h2i88i6A/265-introduce-a-confirmation-dialog-for-the-several-flavors-of-the-accept-request-button

**Comment by danidoni** [comment]
Codeclimate is complaining about some code coverage in a controller that this PR didn't even touch, skipping...

**Comment by hramrach** [comment]
This looks like really atrocious UX to me.

First off, it seems that triangle does not always load (maybe when there is only one option or some error happened).

Second, the 'forward to factory' is still a checkbox. So now there is one option that is a checkbox, and another option that is a menu item, two ways to do the same, but forced to be different by the UI.

With two checkboxes there are actually 4 option, not two. Accept, accept and forward, accept and make maintainer, accept and forward and make maintainer. This is not something that a menu can nicely capture.

Finally, while the other options can readily be activated by single click 'Accept' now requires going though a menu, especially frustrating when there is only one option in that menu. There would be the option to collapse the menu to a simple button when there is only one option but that's also horrible because then the button functionality would be randomly changing.

**Pull request merged**